### ISSUE TITLE
Problems with bash completion autocomplete

### ISSUE THREAD

**Issue by ExplodingDragon** [kanidm/kanidm#2280]
### I did this 

I compiled it using the following command :

```bash
export KANIDM_BUILD_PROFILE="release_suse_generic"
cargo build --locked --release --target-dir target
cp target/release/build/completions/kanidm.bash /usr/share/bash-completion/completions/kanidm
# exit 
# login   
```

When I log out and log back in, press the tab key does not prompt correctly.

This problem also exists in v1.1.0-beta.13.

[![asciicast](https://asciinema.org/a/618341.svg)](https://asciinema.org/a/618341)

### Kanidm version details

- Output of `kanidm(d) version`:  [v1.1.0-rc.15](https://github.com/kanidm/kanidm/releases/tag/v1.1.0-rc.15)

![图片](https://github.com/kanidm/kanidm/assets/33776693/b1bc3fb1-dcf7-4baf-a8b0-8d7838ddc3c4)

**Comment by yaleman** [comment]
If you do `source <completions file>` does it throw any errors? 

Testing it on bash 3.x it throws an error, but 5.x it loads, and completes the first "layer" and then doesn't expand from there.

**Comment by ExplodingDragon** [comment]
> If you do `source <completions file>` does it throw any errors?
> 
> Testing it on bash 3.x it throws an error, but 5.x it loads, and completes the first "layer" and then doesn't expand from there.

No problems with output.
![图片](https://github.com/kanidm/kanidm/assets/33776693/31ad305b-f1fa-4621-8153-0e7519d573ca)

**Comment by ExplodingDragon** [comment]
This is my completion file, using RockyLinux 9. 

[kanidm.bash.gz](https://github.com/kanidm/kanidm/files/13225013/kanidm.bash.gz)

**Comment by yaleman** [comment]
The 3.x "invalid nosort option" error is noted here in clap's repo: https://github.com/clap-rs/clap/issues/5190

It seems to be just the subcommands that fail, and only in bash (zsh works OK). It's an issue with `clap_complete` as far as I can tell, rather than something we have much control over 🥴

**Comment by yaleman** [comment]
> This is my completion file , use RockyLinux 9, compiled with rust v1.73.0 gnu-x86_64.
> 
> [kanidm.bash.gz](https://github.com/kanidm/kanidm/files/13225013/kanidm.bash.gz)

Thanks, matches the file I get here on my test build, so it's not a build environment thing - looks to be some kind of issue coming from `clap`'s generator since I've checked we [match their example](https://docs.rs/clap_complete/latest/clap_complete/generator/fn.generate_to.html#examples).

**Comment by ExplodingDragon** [comment]
https://github.com/clap-rs/clap/issues/5190 has been fixed.

**Comment by ExplodingDragon** [comment]
This issue has been resolved in the latest version (v1.1.0-rc.16).
